File: go/prime.go
package main

type Sieve struct {
	prime int
	prev  *Sieve
}

func Test(payload *Sieve, n int) {
	if payload.prime*(n/payload.prime) == n {

	} else if payload.prev == nil {
		println(n)

		p := *payload
		*payload = Sieve{
			prime: n,
			prev:  &p,
		}
	} else {
		Test(payload.prev, n)
	}
}

func main() {
	var p = Sieve{
		prev:  nil,
		prime: 2,
	}

	println(2)

	for i := 2; i < 10; i++ {
		Test(&p, i)
	}
}

File: rust/prime.rs
struct Sieve {
    prime: i32,
    next :  Option<Box<Sieve>>
}

fn test(payload :&Sieve, n: i32) {
    if (payload.prime * (n / payload.prime) == n) {
        
    } else if (payload.next == nil) {
    
    } else {
        test(payload.next, n)
    }
}

fn main() {
    let mut payload = Sieve { 
        prime: 2
    }

    println!("2")
}